### PR TITLE
Fix missing favorite files in filesystem dock

### LINKED ISSUE #59933: Favorites in the FileSystem panel do not work
### Godot version

Godot 3.5 beta 3

### System information

Windows 10

### Issue description

The Favorites section in the FileSystem panel remains empty.

### Steps to reproduce

There are two cases:

- Open a project on Godot 3.5 beta 3, and use "Add to Favorites" in the FileSystem panel. Observe the Favorites section remains empty.

- Open a project on another version of Godot (e.g. 3.4.4), use "Add to Favorites" in the FileSystem panel. Observe it is added to the Favorites section. Then close the project, and open it on Godot 3.5 beta 3. Observe the Favorites section is empty. Close the project again, and open it on the other version of Godot again. Observe the Favorites section is empty.

### Minimal reproduction project

Any project will do.

### PR DESCRIPTION
**Pull request by DanielKriz**
This pull request addresses #59933.

*Bugsquad edit:*
- Fixes #59933

### CODE REVIEW

File: editor/filesystem_dock.cpp [hunk 1]
```diff
@@ -228,10 +228,10 @@ void FileSystemDock::_update_tree(const Vector<String> &p_uncollapsed_paths, boo
 	DirAccessRef da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
 	bool fav_changed = false;
 	for (int i = favorite_paths.size() - 1; i >= 0; i--) {
-		if (!da->dir_exists(favorite_paths[i])) {
-			favorite_paths.remove_at(i);
-			fav_changed = true;
-		}
+		if (da->dir_exists(favorite_paths[i]) || da->file_exists(favorite_paths[i]))
+			continue;
```
Comment (YeldhamDev) [inline editor/filesystem_dock.cpp:232]:
```suggestion
		if (da->dir_exists(favorite_paths[i]) || da->file_exists(favorite_paths[i])) {
			continue;
		}
```
Single-line `if`s also use brackets in Godot.

**Review by YeldhamDev — CHANGES_REQUESTED** [review]

File: editor/filesystem_dock.cpp (hunk 1 above)
Comment (DanielKriz) [reply editor/filesystem_dock.cpp]: It is adjusted now, thank you.

**Review by YeldhamDev — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by theraot** [comment]
Any chance of this in Godot 3.5?

**Comment by YeldhamDev** [comment]
Looking at this again, it seems that it can be directly cherry-picked.

CC @akien-mga

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Pull request merged**